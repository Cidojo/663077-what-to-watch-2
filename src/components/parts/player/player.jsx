import * as React from 'react';
import * as PropTypes from 'prop-types';

import {Video} from './../video/video.jsx';
import {PlayerControls} from './../player-controls/player-controls.jsx';
import {Button} from './../button/button.jsx';
import withPlayingState from './../../../hocs/with-playing-state/with-playing-state.jsx';
import {movieCardPropTypes} from './../../../global-custom-types.js';

const Player = React.forwardRef(function Player(props, ref) {
  const {
    card,
    muted,
    controls,
    onClosePlayer,
    isPlaying,
    onFullscreenButtonClick,
    onPlayToggle,
    secondsPlayed,
    duration
  } = props;

  const {previewVideoLink, videoLink, posterImage, previewImage, name} = card;

  return (
    <>
      <Video
        videoLink={controls ? videoLink : previewVideoLink}
        posterImage={controls ? posterImage : previewImage}
        muted={muted}
        ref={ref}
      />
      {controls &&
        <>
          <Button
            className="player__exit"
            onClick={onClosePlayer}
          >
            Exit
          </Button>
          <PlayerControls
            movieTitle={name}
            secondsPlayed={secondsPlayed}
            duration={duration}
            isPlaying={isPlaying}
            onPlayToggle={onPlayToggle}
            onFullscreenButtonClick={onFullscreenButtonClick}
          />
        </>
      }
    </>
  );
});

Player.propTypes = {
  secondsPlayed: PropTypes.number,
  duration: PropTypes.number,
  card: movieCardPropTypes,
  muted: PropTypes.bool,
  controls: PropTypes.bool,
  onClosePlayer: PropTypes.func,
  isPlaying: PropTypes.bool,
  onFullscreenButtonClick: PropTypes.func,
  onPlayToggle: PropTypes.func
};

Player.defaultProps = {
  secondsPlayed: 0,
  duration: 0,
  card: {},
  muted: false,
  controls: false,
  onClosePlayer: () => {},
  isPlaying: false,
  onFullscreenButtonClick: () => {},
  onPlayToggle: () => {}
};

export {Player};
export default withPlayingState(Player);
