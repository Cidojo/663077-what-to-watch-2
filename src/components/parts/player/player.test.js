import * as React from 'react';
import {Player} from './player.jsx';
import * as renderer from 'react-test-renderer';

const props = {
  secondsPlayed: 0,
  duration: 0,
  card: {},
  muted: false,
  controls: false,
  onClosePlayer: () => {},
  isPlaying: false,
  onFullscreenButtonClick: () => {},
  onPlayToggle: () => {}
};

it(`should render Player component without errors`, () => {
  const tree = renderer.create(<Player {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
