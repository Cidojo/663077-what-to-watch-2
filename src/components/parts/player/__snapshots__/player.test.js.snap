// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Player component without errors 1`] = `
<video
  className="player__video"
  muted={false}
  poster=""
  src=""
/>
`;
